TensorLib/TensorData.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: John Tristan, Paul Govereau, Sean McLaughlin
+-/
+import Mathlib.Tactic
+import TensorLib.NonEmptyList
+import TensorLib.TensorElement
+
+namespace TensorLib
+
+/-!
+This file defines TensorData, the raw data underlying a Tensor.
+We are representing it today as nested linked lists. We predict
+this will be easier to reason about, but will be slow to evaluate.
+We intend to have a more compact and faster-to-access representation
+in addition to this one in the future for execution.
+-/
+
+-- I tried making this implicit with {} but I needed to add @ everwhere
+-- and give it the argument anyway. Maybe there is a way to infer at least
+-- within a section
+variable (a: Type)
+
+inductive TensorData where
```

Comment:
Will probably be useful eventually to derive useful instantiations such as Repr

Reply:
Yes, implemented `add` yesterday and wanted to print a tensor. That led me to figure out I needed to derive `Repr`. Any others you recommend?